### PR TITLE
TY: take into account type arguments when projecting associated types

### PR DESCRIPTION
**Pull request by vlad20012**
Part of #1806

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/presentation/Ty.kt [hunk 1]
```diff
@@ -61,7 +61,9 @@ private fun render(
         }"
         is TyPointer -> "*${if (ty.mutability.isMut) "mut" else "const"} ${r(ty.referenced)}"
         is TyTypeParameter -> ty.name ?: anonymous
-        is TyProjection -> "<${ty.type} as ${ty.trait.name}>::${ty.target.name}"
+        is TyProjection -> "<${ty.type} as ${ty.trait.element.name ?: return anonymous}${
+        if (includeTypeArguments) formatTraitTypeArguments(ty.trait, r, includeAssoc = false) else ""
+        }>::${ty.target.name}"
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/presentation/Ty.kt]: Maybe split this line into some separate statements?
It's rather hard to understand what happens - too many operations per line
Also a little bit weird formatting

Comment (vlad20012) [reply src/main/kotlin/org/rust/ide/presentation/Ty.kt]: There are other branches that look as awful. This function needs to be cleaned up. But we can do it in another PR.

Comment (Undin) [reply src/main/kotlin/org/rust/ide/presentation/Ty.kt]: Okay:)

**Pull request merged**